include/NeoFOAM/core/database/oldTimeCollection.hpp

```diff
@@ -0,0 +1,159 @@
+ *
+ * This function retrieves the old time field of a given field by looking up the field's name and
+ * time index in the FieldCollection.
+ *
+ * @param field The field to retrieve the old time field from.
+ * @return The old time field.
+ */
+template<typename geoField>
+geoField& oldTime(geoField& field)
+{
+    FieldCollection& fieldCollection = FieldCollection::instance(field);
+    FieldDocument& fieldDoc = fieldCollection.fieldDoc(field.key);
+
+    std::size_t timeIdx = fieldDoc.timeIndex();
+
+    std::string oldTimeName = field.name + "_0";
+
+    std::vector<std::string> oldKeys = fieldCollection.find(
+        [oldTimeName, timeIdx](const Document& doc)
+        {
+            return doc.get<std::string>("name") == oldTimeName
+                && doc.get<std::size_t>("timeIndex") == timeIdx - 1;
+        }
+    );
+
+    bool found = (oldKeys.size() == 1);
+    // print oldKeys
+    for (auto key : fieldCollection.sortedKeys())
+    {
+    }
+    for (auto key : oldKeys)
+    {
+        std::cout << "  -- " << key << std::endl;
+    }
+    if (found)
+    {
+        std::cout << "Found old field" << std::endl;
+        FieldDocument& oldDoc = fieldCollection.fieldDoc(oldKeys[0]);
+        geoField& oldField = oldDoc.field<geoField>();
+        return oldField;
+    }
+    std::cout << "not Found old field" << std::endl;
+
+    // create oldTime field
+    geoField& oldField = fieldCollection.registerField<geoField>(CreateFromExistingField<geoField> {
+        .name = oldTimeName,
+        .field = field,
+        .timeIndex = timeIdx - 1,
+        .iterationIndex = fieldDoc.iterationIndex(),
+        .subCycleIndex = fieldDoc.subCycleIndex()
+    });
+    return oldField;
+}
+
+/**
+ * @brief Retrieves the old time field of a given field (const version).
+ *
+ * This function retrieves the old time field of a given field by looking up the field's name and
+ * time index in the FieldCollection.
+ *
+ * @param field The field to retrieve the old time field from.
+ * @return The old time field.
+ */
+template<typename geoField>
+const geoField& oldTime(const geoField& field)
+{
+    const FieldCollection& fieldCollection = FieldCollection::instance(field);
+    const FieldDocument& fieldDoc = fieldCollection.fieldDoc(field.key);
+
+    std::size_t timeIdx = fieldDoc.timeIndex();
+
+    std::string oldTimeName = field.name + "_0";
+
+    std::vector<std::string> oldKeys = fieldCollection.find(
+        [oldTimeName, timeIdx](const Document& doc)
+        {
+            return doc.get<std::string>("name") == oldTimeName
+                && doc.get<std::size_t>("timeIndex") == timeIdx - 1;
+        }
+    );
+
+    bool found = (oldKeys.size() == 1);
+    if (found)
+    {
+        const FieldDocument& oldDoc = fieldCollection.fieldDoc(oldKeys[0]);
+        const geoField& oldField = oldDoc.field<geoField>();
+        return oldField;
+    }
+    else
+    {
+        NF_THROW("Old field not found");
+    }
```

Comment:
```suggestion
   NF_ASSERT(found,  "Old field, " + oldTimeName + ", not found");
   const FieldDocument& oldDoc = fieldCollection.fieldDoc(oldKeys[0]);
   const geoField& oldField = oldDoc.field<geoField>();
   return oldField;
```

@greole 

In general, do we prefer assertions over throwing (my understanding is that its more performant to abort).